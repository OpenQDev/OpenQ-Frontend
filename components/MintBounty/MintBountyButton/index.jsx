// Third party
import React, { useState } from 'react';
// Custom
import MintBountyModal from '../MintBountyModal/MintBountyModal';
import MintProvider from '../MintProvider';

const MintBountyButton = ({ styles, types }) => {
  const [showModal, setShowModal] = useState(false);

  return (
    <>
      <button
        onClick={() => setShowModal(true)}
        className={`lg:col-start-4 col-span-4 lg:col-span-1 whitespace-nowrap btn-primary flex flex-row space-x-3 items-center justify-center leading-tight h-8 sm:w-min px-3 ${styles}`}
      >
        <svg
          xmlns='http://www.w3.org/2000/svg'
          className='h-5 w-5'
          fill='none'
          viewBox='0 0 24 24'
          stroke='currentColor'
          strokeWidth='2'
        >
          <path
            strokeLinecap='round'
            strokeLinejoin='round'
            d='M9 12h6m-6 4h6m2 5H7a2 2 0 01-2-2V5a2 2 0 012-2h5.586a1 1 0 01.707.293l5.414 5.414a1 1 0 01.293.707V19a2 2 0 01-2 2z'
          />
        </svg>
        <div>New Contract</div>
      </button>
      {showModal && (
        <MintProvider type={types.filter((type) => type != 2)[0]}>
          <MintBountyModal modalVisibility={setShowModal} />
        </MintProvider>
      )}
    </>
  );
};

export default MintBountyButton;
