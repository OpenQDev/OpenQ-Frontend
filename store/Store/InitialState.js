import Utils from '../../services/utils/Utils';

import OpenQClient from '../../services/ethers/OpenQClient';
import OpenQSubgraphClient from '../../services/subgraph/OpenQSubgraphClient';
import AuthService from '../../services/auth/AuthService';
import GithubRepository from '../../services/github/GithubRepository';
import TokenClient from '../../services/coins/TokenClient';
import Logger from '../../services/logger/Logger';
import OpenQPrismaClient from '../../services/openq-api/OpenQPrismaClient';
/* import SuperfluidClient from '../../services/SuperfluidClient/SuperfluidClient'; */

import MockGithubRepository from '../../services/github/MockGithubRepository';
import MockOpenQClient from '../../services/ethers/MockOpenQClient';
/* import MockOpenQSubgraphClient from '../../services/subgraph/MockOpenQSubgraphClient'; */
import MockTokenClient from '../../services/coins/MockTokenClient';
import MockOpenQPrismaClient from '../../services/openq-api/MockOpenQPrismaClient';
import MockAuthService from '../../services/auth/MockAuthService.js';

let InitialState = {};
switch (process.env.NEXT_PUBLIC_DEPLOY_ENV) {
  case 'local':
    InitialState = {
      accountData: {},
      openQClient: new MockOpenQClient(),
      authService: new MockAuthService(),
      githubRepository: new MockGithubRepository(),
      /*  openQSubgraphClient: new MockOpenQSubgraphClient(), */
      tokenClient: new MockTokenClient(),
      logger: new Logger('DEV'),
      utils: new Utils(),
      openQPrismaClient: new MockOpenQPrismaClient(),
      /*  superfluidClient: new SuperfluidClient(), */
    };
    break;
  case 'docker':
    InitialState = {
      accountData: {},
      openQClient: new OpenQClient(),
      authService: new AuthService(),
      githubRepository: new GithubRepository(),
      openQSubgraphClient: new OpenQSubgraphClient(),
      tokenClient: new TokenClient(),
      logger: new Logger('DEV'),
      utils: new Utils(),
      openQPrismaClient: new OpenQPrismaClient(),
      /*  superfluidClient: new SuperfluidClient(), */
    };
    break;
  case 'development':
    InitialState = {
      accountData: {},
      openQClient: new OpenQClient(),
      authService: new AuthService(),
      githubRepository: new GithubRepository(),
      openQSubgraphClient: new OpenQSubgraphClient(),
      tokenClient: new TokenClient(),
      logger: new Logger('DEV'),
      utils: new Utils(),
      openQPrismaClient: new OpenQPrismaClient(),
      /*  superfluidClient: new SuperfluidClient(), */
    };
    break;
  case 'staging':
    InitialState = {
      accountData: {},
      openQClient: new OpenQClient(),
      authService: new AuthService(),
      githubRepository: new GithubRepository(),
      openQSubgraphClient: new OpenQSubgraphClient(),
      tokenClient: new TokenClient(),
      logger: new Logger('DEV'),
      utils: new Utils(),
      openQPrismaClient: new OpenQPrismaClient(),
      /*  superfluidClient: new SuperfluidClient(), */
    };
    break;
  case 'production':
    InitialState = {
      accountData: {},
      openQClient: new OpenQClient(),
      authService: new AuthService(),
      githubRepository: new GithubRepository(),
      openQSubgraphClient: new OpenQSubgraphClient(),
      tokenClient: new TokenClient(),
      logger: new Logger('PROD'),
      utils: new Utils(),
      openQPrismaClient: new OpenQPrismaClient(),
      /*  superfluidClient: new SuperfluidClient(), */
    };
    break;
  default:
    throw Error('ENVIRONMENT NOT CONFIGURED CORRECTLY. Set an environment with DEPLOY_ENV');
}

export default InitialState;
