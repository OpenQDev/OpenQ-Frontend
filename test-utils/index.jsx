// Third party
import React from 'react';
// Cusotm
import { render, renderHook } from '@testing-library/react';
import { cleanup } from '@testing-library/react';

import StoreProvider from '../store/Store/TestStoreProvider';
import AuthProvider from '../store/AuthStore/TestAuthProvider';
import FundProvider from '../components/FundBounty/FundStore/TestFundProvider';
import TokenProvider from '../components/TokenSelection/TokenStore/TokenProvider';

// Add in any providers here if necessary:
// test-utils.js
// (ReduxProvider, ThemeProvider, etc)
afterEach(() => {
  cleanup();
});

const customRender = (ui, options = {}, storeProps, authProps) => {
  const IntersectionObserverMock = vi.fn(() => ({
    disconnect: vi.fn(),
    observe: vi.fn(),
    takeRecords: vi.fn(),
    unobserve: vi.fn(),
  }));
  vi.stubGlobal('IntersectionObserver', IntersectionObserverMock);
  const Providers = ({ children }) => {
    return (
      <AuthProvider StoreProps={authProps}>
        <StoreProvider StoreProps={storeProps}>
          <TokenProvider>{children}</TokenProvider>
        </StoreProvider>
      </AuthProvider>
    );
  };

  return render(ui, { wrapper: Providers, ...options });
};
const customHookRender = (hook, options = {}, storeProps, authProps, fundProps) => {
  const Providers = ({ children }) => {
    return (
      <AuthProvider StoreProps={authProps}>
        <StoreProvider StoreProps={storeProps}>
          <FundProvider StoreProps={fundProps}>
            <TokenProvider>{children}</TokenProvider>
          </FundProvider>
        </StoreProvider>
      </AuthProvider>
    );
  };

  return renderHook(hook, { wrapper: Providers, ...options });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, customHookRender as hookRender };
