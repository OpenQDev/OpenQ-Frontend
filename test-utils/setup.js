import { expect, beforeEach } from 'vitest';
import { cleanup } from '@testing-library/react';
import matchers from '@testing-library/jest-dom/matchers';

expect.extend(matchers);
beforeEach(() => {
  cleanup();
});
process.on('unhandledRejection', (reason) => {
  // eslint-disable-next-line no-console
  console.log(`FAILED TO HANDLE PROMISE REJECTION`);
  throw reason;
});
