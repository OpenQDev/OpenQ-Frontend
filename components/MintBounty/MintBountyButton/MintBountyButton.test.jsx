/**
 * @vi-environment jsdom
 */
import React from 'react';
import { render, screen } from '../../../test-utils';
import MintBountyButton from '../MintBountyButton';
import userEvent from '@testing-library/user-event';
import InitialState from '../../../store/Store/InitialState';
import renderer from 'react-test-renderer';
import nextRouter from 'next/router';

InitialState.openQClient.shouldSleep = 200;
nextRouter.useRouter = vi.fn();
nextRouter.useRouter.mockImplementation(() => ({
  query: { type: null },

  prefetch: vi.fn(() => {
    return { catch: vi.fn };
  }),
}));

describe('MintBountyButton', () => {
  it('should match DOM Snapshot', () => {
    const tree = renderer.create(<MintBountyButton />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('should open MintBountyModal on fixed price', async () => {
    // ARRANGE
    const user = userEvent.setup();
    render(<MintBountyButton types={['0', '1', '2', '3']} />);

    // ACT
    const mintBountyButton = await screen.findByRole('button', { name: /New Contract/i });
    await user.click(mintBountyButton);
    expect(await screen.findByText(/Deploy Fixed Price Contract/i)).toBeInTheDocument();
    expect(screen.getByText(/Create a Fixed Price Contract to send funds to any GitHub issue/i));
  });

  it('should open MintBountyModal on contest', async () => {
    // ARRANGE
    const user = userEvent.setup();
    render(<MintBountyButton types={['2', '3']} />);

    // ACT
    const mintBountyButton = await screen.findByRole('button', { name: /New Contract/i });
    await user.click(mintBountyButton);
    expect(await screen.findByText(/Deploy Hackathon Contract/i)).toBeInTheDocument();
    expect(await screen.findByText(/New Contract/i)).toBeInTheDocument();
    expect(screen.getByText(/How many tiers/i));
  });
});
