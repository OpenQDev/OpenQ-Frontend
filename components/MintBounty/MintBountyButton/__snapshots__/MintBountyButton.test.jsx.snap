// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MintBountyButton > should match DOM Snapshot 1`] = `
<button
  className="lg:col-start-4 col-span-4 lg:col-span-1 whitespace-nowrap btn-primary flex flex-row space-x-3 items-center justify-center leading-tight h-8 sm:w-min px-3 undefined"
  onClick={[Function]}
>
  <svg
    className="h-5 w-5"
    fill="none"
    stroke="currentColor"
    strokeWidth="2"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9 12h6m-6 4h6m2 5H7a2 2 0 01-2-2V5a2 2 0 012-2h5.586a1 1 0 01.707.293l5.414 5.414a1 1 0 01.293.707V19a2 2 0 01-2 2z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div>
    New Contract
  </div>
</button>
`;
